### ISSUE TITLE
Cannot set enumeration for `nullptr` values

### ISSUE THREAD

**Issue by op2786** [Vector35/binaryninja-api#4117]
**Version and Platform (required):**
- Binary Ninja Version: 3.4.4133-dev
- OS: Windows 11, x64

**Bug Description:**
See the GIF please. Maybe this is because I try to set enum for `nullptr` value? Even if it because of `nullptr`, I believe that this shouldn't be happening.

**Expected Behavior:**
I expect it to set `nullptr` to my enum or do not open select enum dialog for `nullptr` values.

**Screenshots:**
![Recording 2023-03-17 at 08 39 20](https://user-images.githubusercontent.com/109607896/225842652-bfc27110-fd2a-4947-b55d-f91f7501ac8d.gif)

**Comment by psifertex** [comment]
I wonder what happens if you use "y" to change the function parameter type instead of the UI since it looks like it's misaligned. Also, does it behave the same in HLIL as well?

**Comment by op2786** [comment]
If I change 4th parameter type to `int` (which was `PSTR` before) and then select enum again this is what happens:

![Screenshot 2023-03-17 090404](https://user-images.githubusercontent.com/109607896/225848498-0b266974-3b86-4ff8-a0b6-6f8711b081fe.png)

It set enumeration value to `CC_FASTCALL` for both 3rd and 4th parameter. Which is also does not correct, because I just want to set it for 4th parameter. `CC_FASTCALL` is meaningless here, I just use it in order to demonstrate the bug.

I'm not sure what you mean by HLIL. The GIF above is from HLIL output. If you mean what happens for examples like below, it does not change value to enum, it left it as `nullptr`.

```
004174e9  void* var_650 = nullptr
004174ed  void var_648
```

**Comment by psifertex** [comment]
Are you sure the gif isn't pseudo c? It shows casts in C style and it doesn't have names on the parameters.

**Comment by op2786** [comment]
Yes I'm sure. I disabled names for parameters. Pseudo-C always have `;` in the end of statements.

**Comment by op2786** [comment]
I guess this is for all nullptr values. I can reproduce it for all nullptr values that I saw on function call. Let me show some examples:

Let's say I've this:

```
function(a, 0, nullptr, b);
```

When I try to select enum for third parameter, actually it sets for second parameter. 

```
function2(a, nullptr, b);
```

For the example above, when I try to set enum for second parameter, nothing happens.

Another example:

```
function3(1, 1, 6, nullptr, 0, 0)
```

When I try to set enum for fourth parameter here, it sets enum for fiveth and sixth parameter.

When I try to set enum for nullptr variables (`var114 = nullptr` ), again, nothing happens.
